draft/0005-impersonation.md

```diff
@@ -0,0 +1,279 @@
+
+### `client_id`
+
+A JWT access token in Logto contains a `client_id` claim that represents the client application that requested the token. When a subject token is issued, the `client_id` claim in the subject token is set to the `client_id` of the machine-to-machine application that requested the subject token.
+
+## Example
+
+The following example demonstrates how a support engineer can use the user impersonation feature to troubleshoot an issue on behalf of a user. Let's say the support engineer received a report from a user that they are unable to access a specific resource. The support engineer needs to investigate the issue and verify the user's permissions.
+
+The following sequence diagram illustrates the flow of the user impersonation feature in Logto.
+
+```mermaid
+sequenceDiagram
+    participant SE as Support Engineer
+    participant RO as Resource Owner API
+    participant M2M as Logto Management API
+    participant TS as Tenant's Token Endpoint
+    participant RS as Resource
+    
+    SE ->> RO: Request for subject token
+    RO ->> RO: validate user permission
+    RO ->> M2M: Request for subject token
+    M2M ->> RO: subject token
+    RO ->> SE: subject token
+    
+    SE ->> TS: POST /oidc/token to grant
+    TS ->> SE: access_token (along with other params)
+    
+    SE ->> RS: Fetch user resource with the access token
+    RS ->> SE: Return data
+```
+
+### Step 1: Issue a subject token
+
+The support engineer calls the resource owner's API to issue a subject token for the user who reported the issue. The following request is a simple example of the resource owner's API, the resource owner MUST implement the necessary security controls to validate the support engineer's request according to their business requirements.
+
+```
+POST /request-for-subject-token HTTP/1.1
+Host: resource.example.com
+Authorization: Bearer ...
+Content-Type: application/json
+
+{
+  "userId":"123456",
+  "ticketId":"7890"
+}
+```
+
+The resource owner API validates the support engineer's request and call Logto's management API using a machine-to-machine application to issue a subject token for the user.
+
+```
+POST /api/security/subject-tokens HTTP/1.1
+Host: example.logto.app
+Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9...
+Content-Type: application/json
+
+{
+  "userId":"123456",
+  "expiresIn":600
+}
+```
+
+Logto validates the request and issues a subject token for the user.
+
+```
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+  "subjectToken":"accVkjcJyb4BWCxGsndESCJQbdFMogUC5PbRDqceLTC",
+  "expiresIn":600
+}
+```
+
+The resource owner API returns the subject token to the support engineer.
+
+### Step 2: Exchange the subject token for an access token
+
+The support engineer exchanges the subject token for an access token that represents the identity of the user. The support engineer can use the access token to make API requests on behalf of the user.
```

Comment:
since there's no access token yet:

```suggestion
The support engineer exchanges the subject token for an access token that represents the identity of the user:
```